src/coreComponents/unitTests/constitutiveTests/testCO2BrinePVTModels.cpp

```diff
@@ -1285,12 +1265,12 @@ TEST_F( CO2InternalEnergyTest, CO2InternalEnergyMassValuesAndDeriv )
                                       12984.500000, 12985.100000, 12985.600000};
 
 
-  localIndex counter = 0;
-  for( localIndex iComp = 0; iComp < 3; ++iComp )
+  integer counter = 0;
+  for( integer iComp = 0; iComp < 3; ++iComp )
   {
-    for( localIndex iPres = 0; iPres < 3; ++iPres )
+    for( integer iPres = 0; iPres < 3; ++iPres )
     {
-      for( localIndex iTemp = 0; iTemp < 3; ++iTemp )
+      for( integer iTemp = 0; iTemp < 3; ++iTemp )
```

Comment:
It's nice that you are changing these `localIndex` into `integer` but I do think that you are wasting your energy.
If we really want to do this, then a carefully written `search`/`replace` on the whole code base will do the duty.

Reply:
I heard the discussion this morning about `integer` loops but I still do not fully understand. Take this loop
```c++
for( integer ip = 0; ip < numPhase; ++ip )
{
  phaseMolecularWeight[ip] = props.getMolecularWeight( m_phaseTypes[ip] ).value;
  phaseMolecularWeight[ip][Deriv::dP] = props.getMolecularWeight( m_phaseTypes[ip] ).dP;
  [...]
```
I thing the indexing type of `phaseMolecularWeight` or `phaseMolecularWeight` is `localIndex`. So what is the point at using `integer` instead of `phaseMolecularWeight::IndexType` which appear to be a `localIndex`? Performances really get improved there?

Reply:
For me this is not about performance but semantics. A `localIndex` is a counterpart to `globalIndex` where "local" and "global" refer, generally speaking, to parallel partition of the physical domain and mesh. Therefore these terms make sense when indexing elements, nodes, faces, etc which can be either local or global. Model properties like number of phases or components are in a different category - they are much smaller, and the notion of local/global does not apply to them. Ideally we could introduce separate aliases for them, e.g.
```
using phaseIndex = /* some type */;
using componentIndex = /* some type */;
```
but I thought this might be a bit overkill. Therefore I started changing, and suggesting to others in reviews (mainly @francoishamon who did most of the work), those indices to a generic `integer`. As a secondary concern, we also get to save some bytes in kernel wrapper objects, which sometimes need to carry phase/component indices as members. But it's a really minor saving compared to what this PR achieves at scale.

Now, `phaseMolecularWeight::IndexType` is indeed currently a `localIndex`. But that is also wrong in my opinion! For certain "small" arrays (e.g. `m_componentMolecularWeight`) we could be using a smaller type such as `integer` as index type, and it would also help a bit with reducing the size of kernel arguments. I don't advocate necessarily that we implement this change, because the increasing complexity in the type system has cognitive costs (there would be `array1d<T>` and some `array1dSmallIndex<T>` to keep in mind). But neither am I concerned about integer promotions taking place, just like I'd never force someone to change
```c++
std::vector< double > v;
v[0] ...
```
to
```c++
v[ std::vector< double >::size_type(0) ] ...
```

Reply:
OK, thanks for the explanations. Some kind of Hungarian stuff on types, in a nutshell?